desktop/cmp/form/NumberField.js

```diff
@@ -114,5 +116,11 @@ export class NumberField extends HoistField {
         return parseFloat(value);
     }
 
+    onNumberFieldFocus = (ev) => {
+        if (this.props.selectOnFocus === true) {
```

Comment:
I think it's worth simplifying these to `if (this.props.selectOnFocus) {}` given that it should be reliably confirmed as a boolean via propTypes (and some other truthy value would not be the worst thing here).

More interestingly, @lbwexler did you and Arjun discuss the rather sticky bit of the superclass declaring `onFocus` as a property assigned to an arrow function (vs. a method that could be called via `super.onFocus()` - but would need to be properly bound for the handlers). 

This was my best suggestion - declare a more specific handler and use that - but I was treating that as somewhat provisional as it seemed like this could generally be a source of mistakes....

Reply:
I'll make the code change for `(this.props.selectOnFocus)` now as a separate commit.

Also, I did not have the discussion with @lbwexler about an arrow function vs. a method for these kinds of properties ... but will today.